modules/test-utils/src/test-renderer.js

```diff
@@ -158,17 +161,21 @@ export default class TestRenderer {
       this.currentTestCase = testCase;
       this._currentTestCaseStartTime = animationProps.time;
       this._currentTestCaseStartTick = animationProps.tick;
+      this._testCaseData = null;
       // initialize test case
       withParameters(animationProps.gl, DEFAULT_RENDER_PARAMETERS, () => {
-        this._testCaseData =
+        Promise.resolve(
```

Comment:
This resolves **immediately**
You can just do `this._testCaseData = ...`

Reply:
Not if `onInitialized` returns a promise.

Reply:
`(testCase.onInitialize() || Promise.resolve({})).then(...)`

Reply:
I think that way is easier to read.

Reply:
That won't work if `onInitialized` returns a plain object.

Reply:
OK, but can you add a comment?